### PR TITLE
dev/core#3739 fixes for payment processor display and search in advanced search

### PR DESCRIPTION
**Pull request by kurund**

### CODE REVIEW

File: CRM/Contribute/DAO/ContributionRecur.php [hunk 1]
```diff
@@ -783,7 +783,7 @@ public static function &fields() {
           'pseudoconstant' => [
             'table' => 'civicrm_payment_processor',
             'keyColumn' => 'id',
-            'labelColumn' => 'title',
+            'labelColumn' => 'name',
```
Comment (demeritcowboy) [inline CRM/Contribute/DAO/ContributionRecur.php:786]: I don't think this is the right fix. `title` is the right label column. There's multiple problems here I think. One is that `title` is optional, unlike other entities. The other is `name` is exposed on the form and is used as the user-visible identifier, so `title` is often left blank. I'm not sure if there's a quick-fix. The `title` could be auto-computed to be the same as `name` if blank? Or maybe like other entities `name` would be hidden on the form and auto-computed from `title` for new records, and `title` would be required on the form.

Comment (kurund) [reply CRM/Contribute/DAO/ContributionRecur.php:786]: > One is that title is optional, unlike other entities.

Yes, hence it make sense to display what's required

> The other is name is exposed on the form and is used as the user-visible identifier, so title is often left blank. 

I feel the drop down should display what's seen by the user in other places hence the fix.

> The title could be auto-computed to be the same as name if blank? Or maybe like other entities name would be hidden on the form and auto-computed from title for new records, and title would be required on the form.

I think this is a unique case where in `name` serves as `title` unlike other places where name is not exposed in the UI. May be we should deprecate 'title'?

Comment (kurund) [reply CRM/Contribute/DAO/ContributionRecur.php:786]: Do you think this patch could be interim solution till we deprecate title?

Comment (mattwire) [reply CRM/Contribute/DAO/ContributionRecur.php:786]: `title` is what the end-user should/will see. `name` is what the admin should see.  The code should do this in most places: `$displayname = $paymentProcessor['title'] ?? $paymentProcessor['name'];`

Eg. you have 2 stripe payment processors: 1. `name` => "Stripe main account", `title` => "Credit Card". 2. `name` => "Stripe charity account", `title` => "Credit Card".

So the end user just sees "Credit Card" on the appropriate contribution page but the admin knows which one it is. For a search form you probably want to see the name or ideally both but changing the labelColumn attribute will almost certainly have wider, unintended consequences.

**Comment by kurund** [comment]
Based on the above discussion I am closing this PR.

A workaround is to always have a title for the payment processor so that it's displayed in Advanced search.

**Pull request closed without merging**